processing/src/main/java/io/druid/segment/IndexSpec.java

```diff
@@ -91,7 +94,8 @@ public IndexSpec()
   public IndexSpec(
       @JsonProperty("bitmap") BitmapSerdeFactory bitmapSerdeFactory,
       @JsonProperty("dimensionCompression") String dimensionCompression,
-      @JsonProperty("metricCompression") String metricCompression
+      @JsonProperty("metricCompression") String metricCompression,
+      @JsonProperty("metricLongEncoding") String metricLongEncoding
```

Comment:
can we call this metricEncoding to keep the configuration simple? We can simply return an error if someone uses an encoding that doesn't apply to a certain type of metric.